tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+	}
+
+	return false
+}
+
+func ICanRunACyclesStressTestWithReadsAndAOpssRateWithinTimeoutStep(t *testing.T, stressCycles, percentRead string, rate, timeout int) {
+	t.Logf("I run a %s cycles stess test with %s reads and %d ops/s within %d seconds", stressCycles, percentRead, rate, timeout)
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	cqlCredentials := ExtractUsernamePassword(t, "k8ssandra-superuser", namespace)
+	parsedReadRatio, _ := strconv.Atoi(strings.ReplaceAll(percentRead, "%", ""))
+	readRatio := parsedReadRatio / 10
+	writeRatio := 10 - readRatio
+
+	jobName := fmt.Sprintf("nosqlbench-%s", strings.ToLower(random.UniqueId()))
+	k8s.RunKubectl(t, kubectlOptions, "create", "job", "--image=nosqlbench/nosqlbench", jobName,
+		"--", "java", "-jar", "nb.jar", "cql-iot", "rampup-cycles=1k", fmt.Sprintf("cyclerate=%d", rate),
+		fmt.Sprintf("username=%s", cqlCredentials.username), fmt.Sprintf("password=%s", cqlCredentials.password),
+		fmt.Sprintf("main-cycles=%s", stressCycles), "hosts=k8ssandra-dc1-stargate-service", "--progress", "console:1s", "-v",
+		fmt.Sprintf("write_ratio=%d", writeRatio), fmt.Sprintf("read_ratio=%d", readRatio), "async=100")
+
+	defer timeTrack(time.Now(), fmt.Sprintf("nosqlbench stress test with %d ops/s", rate))
+	k8s.RunKubectl(t, kubectlOptions, "wait", "--for=condition=complete", fmt.Sprintf("--timeout=%ds", timeout), fmt.Sprintf("job/%s", jobName))
+
+	output := RunShellCommandAndGetOutput(
+		exec.Command("bash", "-c", fmt.Sprintf("kubectl logs job/%s -n %s | grep -e cqliot_default_main.cycles.servicetime -e cqliot_default_main.cycles.responsetime", jobName, namespace)))
+	t.Log(Outline(output))
+}
+
+func WaitForStargatePodReady(t *testing.T) {
+	g(t).Eventually(func() bool {
+		return stargatePodIsReady(t)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue(), "Stargate deployment didn't roll out within timeout")
+}
+
+func stargatePodIsReady(t *testing.T) bool {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "rollout", "status", "deployment", "k8ssandra-dc1-stargate")
+	if err == nil {
+		if strings.HasSuffix(output, "successfully rolled out") {
+			return true
+		}
+	}
+	return false
+}
+
+func DeployMinioAndCreateBucket(t *testing.T, bucketName string) {
+	helmOptions := &helm.Options{
+		KubectlOptions: k8s.NewKubectlOptions("", "", "default"),
+	}
+	helm.RunHelmCommandAndGetOutputE(t, helmOptions, "repo", "add", "minio", "https://helm.min.io/")
+
+	helm.RunHelmCommandAndGetOutputE(t, helmOptions, "install",
+		"--set", fmt.Sprintf("accessKey=minio_key,secretKey=minio_secret,defaultBucket.enabled=true,defaultBucket.name=%s", bucketName),
+		"--generate-name", "minio/minio", "-n", "minio", "--create-namespace")
+}
+
+func CheckClusterExpectedResources(t *testing.T) {
+	checkResourcePresence(t, "service", "k8ssandra-dc1-all-pods-service")
+	checkResourcePresence(t, "service", "k8ssandra-dc1-service")
```

Comment:
These names are only true if datacenter name can't be modified.

Reply:
Sure, but we have control over the deployment parameters and they are using a fixed value.
It can definitely be parameterized in the future if tests require it.

Reply:
Considering that this is a helper function that can be called from different tests I think it would be best to parameterize it now.

Reply:
Done. The datacenter name (although being a constant) is parameterized everywhere (and if changed will be forcefully set during helm install).
Same goes for the release name.